f8a_version_comparator/item_object.py

```diff
@@ -0,0 +1,219 @@
+        """
+        self.qualifiers = ["alpha", "beta", "milestone", "rc", "snapshot", "", "sp"]
+
+        self.aliases = {
+               "ga": "",
+               "final": "",
+               "cr": "rc"
+        }
+
+        self.release_version_index = str(self.qualifiers.index(""))
+        self._decode_char_versions(str_version, followed_by_digit)
+
+    def _decode_char_versions(self, value, followed_by_digit):
+        """Decode short forms of versions."""
+        if followed_by_digit and len(value) == 1:
+            if value.startswith("a"):
+                value = "alpha"
+            elif value.startswith("b"):
+                value = "beta"
+            elif value.startswith("m"):
+                value = "milestone"
+
+        self.value = self.aliases.get(value, value)
+
+    def comparable_qualifier(self, qualifier):
+        """Get qualifier that is comparable."""
+        q_index = None
+        if qualifier in self.qualifiers:
+            q_index = self.qualifiers.index(qualifier)
+        q_index_not_found = str(len(self.qualifiers)) + "-" + qualifier
+
+        return str(q_index) if q_index is not None else q_index_not_found
+
+    def str_cmp(self, val1, val2):
+        """Compare two strings."""
+        if val1.__lt__(val2):
+            return -1
+        if val1.__gt__(val2):
+            return 1
+        return 0
+
+    def compare_to(self, item):
+        """Compare two maven versions."""
+        if item is None:
+            temp = self.str_cmp(self.comparable_qualifier(self.value), self.release_version_index)
+            return temp
+        if isinstance(item, IntegerItem):
+            return -1
+        if isinstance(item, StringItem):
+            return self.str_cmp(
+                self.comparable_qualifier(
+                    self.value), self.comparable_qualifier(
+                    item.value))
+        if isinstance(item, ListItem):
+            return -1
+        else:
+            raise ValueError("invalid item" + type(item))
+
+    def to_string(self):
+        """Return value in string form."""
+        return str(self.value)
+
+    @classmethod
+    def is_none(self):
+        """Check if none."""
+        if self.value is release_version_index and self.value is 0:
+            return True
+
+        return False
+
+
+class ListItem(Item):
+    """String Item class for maven version comparator tasks."""
+
+    def __init__(self):
+        """Initialize string value of version."""
+        self.array_list = list()
+
+    def add_item(self, item):
+        """Add item to array list."""
+        arr_list = self.array_list
+        arr_list.append(item)
+
+    def get_list(self):
+        """Get object list items."""
+        return self.array_list
+
+    def normalize(self):
+        """Remove trailing items: 0, "", empty list."""
+        red_list = [0, None, ""]
+        i = len(self.array_list) - 1
+        while(i >= 0):
+            lastItem = self.array_list[i]
```

Comment:
We use snake_case everywhere in the code, so camelCase like `lastItem` shouldn't be here.